### PR TITLE
feat: legal disclaimer

### LINKED ISSUE #54: Add Legal Disclaimer to Vaults UI
## What is the Problem Being Solved?
Similar to PSM and Wallet UI, we need to serve users with a legal disclaimer upon accessing the UI for the first time. User must accept to be able to continue
## Description of the Design

Same design as PSM: https://github.com/Agoric/agoric-sdk/issues/6491 . Copy TBD once Legal confirms.

## Security Considerations
N/A
## Test Plan
To be tested during smoke testing

### PR DESCRIPTION
**Pull request by samsiegart**
fixes #54 

image

### CODE REVIEW

File: src/components/ActionsDialog.tsx [hunk 1]
```diff
@@ -61,12 +61,14 @@ const ActionsDialog = ({
                 <div className="h-[1px] mx-8 bg-[#D8D8D8]" />
                 <div className="py-6 px-8">
                   <div className="flex justify-end gap-6">
-                    <button
-                      className="text-btn-xs flex justify-center rounded  text-[#A3A5B9] border-[#A3A5B9] border-2 px-6 py-3 bg-gray-500 bg-opacity-0 hover:bg-opacity-10 active:bg-opacity-20"
-                      onClick={onSecondaryAction}
-                    >
-                      {secondaryActionLabel}
-                    </button>
+                    {secondaryActionLabel && onSecondaryAction && (
```
Comment (turadg) [inline src/components/ActionsDialog.tsx:64]: The type makes them independently optional but this is showing they go together. Consider making the type like `secondaryAction?: [label: string, onPress: () => void]`

File: src/components/TermsDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          checked={isChecked}
+          onClick={() => setIsChecked(isChecked => !isChecked)}
+        />
+        <span>
+          By clicking here you are indicating that you have read and agree to
+          our{' '}
+          <a
+            className="text-blue-500 hover:text-blue-700 underline transition-colors"
+            target="inter_psm_disclaimer"
+            href="https://docs.inter.trade/disclaimer"
```
Comment (turadg) [inline src/components/TermsDialog.tsx:43]: may change independently of behavior. should be a const. maybe in a config module

File: src/components/TermsDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            Disclaimer
+          </a>
+          .
+        </span>
+      </label>
+    </div>
+  );
+
+  return (
+    <>
```
Comment (turadg) [inline src/components/TermsDialog.tsx:54]: unnecessary nesting

File: src/service/wallet.ts [hunk 4]
```diff
@@ -84,6 +89,13 @@ export const makeWalletService = () => {
 
     if (isWalletConnectionInProgress || chainConnection) return;
 
+    if (
+      latestTermsIndex !== localStorageStore.getState().latestTermsAgreedIndex
```
Comment (turadg) [inline src/service/wallet.ts:93]: consider having something in the dialog (maybe in small gray print) that the terms must be agreed to in each browser. to stave off support requests about having to agree repeatedly. 

do we check anywhere that localStorage is available? we might need to error when it's disabled. people may auto-clear it but not much we can do about that

File: src/store/app.ts [hunk 5]
```diff
@@ -134,3 +143,22 @@ export const walletUiHrefAtom = atom(get => {
 
   return bridgeUrl ? bridgeUrl.origin + '/wallet/' : '';
 });
+
+// Increment whenever the terms on https://docs.inter.trade/disclaimer change.
+export const latestTermsIndex = 0;
```
Comment (turadg) [inline src/store/app.ts:148]: "Terms" has a meaning in our contract API. consider "Disclaimer" or "[ToS](https://en.wikipedia.org/wiki/Terms_of_service)"

File: yarn.lock [hunk 6]
```diff
@@ -6643,9 +6643,9 @@ yocto-queue@^0.1.0:
   resolved "https://registry.yarnpkg.com/yocto-queue/-/yocto-queue-0.1.0.tgz#0294eb3dee05028d31ee1a5fa2c556a6aaf10a1b"
   integrity sha512-rVksvsnNCdJ/ohGc6xgPwyN8eheCxsiLM8mxuE/t/mOVqJewPuO1miLpTHQiRgTKCLexL4MeAFVagts7HmNZ2Q==
 
-zustand@^4.1.5:
-  version "4.1.5"
-  resolved "https://registry.yarnpkg.com/zustand/-/zustand-4.1.5.tgz#7402b511f5b23ccb0f9ba6d20ae01ec817e16eb6"
-  integrity sha512-PsdRT8Bvq22Yyh1tvpgdHNE7OAeFKqJXUxtJvj1Ixw2B9O2YZ1M34ImQ+xyZah4wZrR4lENMoDUutKPpyXCQ/Q==
+zustand@^4.3.7:
```
Comment (turadg) [inline yarn.lock:6646]: curious, was this necessary or just tidying up?

for changelog hygiene please try to keep dep bumps in their own commits

**Review by turadg — APPROVED** [review]

File: yarn.lock (hunk 6 above)
Comment (samsiegart) [reply yarn.lock]: Neither, I was just trying to get it to work and it wasn't actually needed. Reverted this change for now.

File: src/service/wallet.ts (hunk 4 above)
Comment (samsiegart) [reply src/service/wallet.ts]: I'm not sure exactly how to word that, especially considering multiple ipfs hash domains, but I can follow up and see if that's more helpful or just more confusing.

Looks like if localstorage isn't available, the app just crashes because `atomWithStorage` throws. I'm reluctant to try and handle that because it's pretty out of the ordinary, and at least for chrome, you'll see a "site not working?" tip whenever it's off by default.

File: src/components/ActionsDialog.tsx (hunk 1 above)
Comment (samsiegart) [reply src/components/ActionsDialog.tsx:64]: Done. I was just hesitant about un-flattening the params because it impacts equality checking, but for this it's not much impact.
